src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -34,35 +38,35 @@ public class ConceptChildrenQueryBuilder {
 
   /**
    * Generate a query that retrieves the descendants of the given concept and their roll-up counts.
-   *
-   * <p>SELECT cc.concept_id, cc.concept_name, COUNT(DISTINCT co.person_id) as count FROM `concept`
-   * AS cc JOIN `concept_ancestor` AS ca ON cc.concept_id = ca.ancestor_concept_id JOIN
-   * `'domain'_occurrence` AS co ON co.'domain'_concept_id = ca.descendant_concept_id WHERE
-   * (c.concept_id IN (SELECT c.concept_id_2 FROM concept_relationship AS c WHERE (c.concept_id_1 =
-   * 101 AND * c.relationship_id = 'Subsumes')) AND c.standard_concept = 'S') GROUP BY
-   * cc.concept_id, cc.concept_name ORDER BY cc.concept_name ASC
    */
   public Query buildConceptChildrenQuery(
       SnapshotBuilderDomainOption domainOption, int parentConceptId) {
 
     // concept table and its fields concept_name and concept_id
-    TableVariable concept = TableVariable.forPrimary(TablePointer.fromTableName(CONCEPT));
+    SourceVariable concept = SourceVariable.forPrimary(TablePointer.fromTableName(CONCEPT));
     FieldVariable conceptName = concept.makeFieldVariable(CONCEPT_NAME);
     FieldVariable conceptId = concept.makeFieldVariable(CONCEPT_ID);
     FieldVariable conceptCode = concept.makeFieldVariable(CONCEPT_CODE);
 
     // concept_ancestor joined on concept.concept_id = ancestor_concept_id.
     // We use concept_ancestor for the rollup count because we want to include counts
     // from all descendants, not just direct descendants.
-    TableVariable conceptAncestor =
-        TableVariable.forJoined(
+    SourceVariable conceptAncestor =
+        SourceVariable.forJoined(
             TablePointer.fromTableName(CONCEPT_ANCESTOR), ANCESTOR_CONCEPT_ID, conceptId);
     FieldVariable descendantConceptId = conceptAncestor.makeFieldVariable(DESCENDANT_CONCEPT_ID);
 
+    var subQuery = createSubQuery(parentConceptId);
+    var subQueryPointer = new SubQueryPointer(subQuery, "join_filter");
```

Comment:
Maybe theres a clearer name for the subquery as well instead of "join_filter" just to help make this more readable